modules/cnf-running-cnf-tests-in-discovery-mode.adoc

```diff
@@ -0,0 +1,274 @@
+
+.DPDK tests
+
+The DPDK related tests require:
+
+* A PerformanceProfile
+* A SRIOV policy
+* A node with resources available for the SRIOV policy and available with the PerformanceProfile node selector
+
+.PTP tests
+
+* A slave PtpConfig (`ptp4lOpts="-s" ,phc2sysOpts="-a -r"`)
+* A node with a label matching the slave PtpConfig
+
+.SCTP tests
+
+* SriovNetworkNodePolicy
+* A node matching both the SriovNetworkNodePolicy and a MachineConfig that enables SCTP
+
+.Performance operator tests
+
+Various tests have different requirements. Some of them are:
+
+* A PerformanceProfile
+* A PerformanceProfile having `profile.Spec.CPU.Isolated = 1`
+* A PerformanceProfile having `profile.Spec.RealTimeKernel.Enabled == true`
+* A node with no huge pages usage
+
+== Limiting the nodes used during tests
+
+The nodes on which the tests are executed can be limited by specifying a `NODES_SELECTOR` environment variable.
+Any resources created by the test are then limited to the specified nodes.
+
+----
+docker run -v $(pwd)/:/kubeconfig:Z -e KUBECONFIG=/kubeconfig/kubeconfig -e
+NODES_SELECTOR=node-role.kubernetes.io/worker-cnf quay.io/openshift-kni/cnf-tests /usr/bin/test-run.sh
+----
+
+== Ginkgo parameters
+
+The test suite is built upon the Ginkgo BDD framework.
+This means that it accepts parameters for filtering or skipping tests.
+
+To filter a set of tests, add the `-ginkgo.focus` parameter:
+
+----
+docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig
+quay.io/openshift-kni/cnf-tests /usr/bin/test-run.sh -ginkgo.focus="performance|sctp"
+----
+
+[NOTE]
+====
+There is a particular test ([sriov] SCTP integration) that requires both SR-IOV and SCTP.
+Given the selective nature of the `focus` parameter, this test is triggered by only placing the `sriov` matcher.
+If the tests are executed against a cluster where SR-IOV is installed but SCTP is not, adding the
+`-ginkgo.skip=SCTP` parameter causes the tests to skip SCTP testing.
+====
+
+=== Available features
+
+The set of available features to filter are:
+
+* performance
+* sriov
+* ptp
+* sctp
+* dpdk
+
+=== Dry run
+
+To run in dry-run mode run the following command:
+
+----
+docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig
+quay.io/openshift-kni/cnf-tests /usr/bin/test-run.sh -ginkgo.dryRun -ginkgo.v
+----
+
+== Reducing test running time
+
+=== Using a single performance profile
+
+The resources needed by the DPDK tests are higher than those required by the performance test suite.
+To make the execution quicker, the performance profile used by tests can be overridden using one that also serves
+the DPDK test suite.
+
+To do this, a profile like the following one can be mounted inside the container,
+and the performance tests can be instructed to deploy it.
+
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v1
+kind: PerformanceProfile
+metadata:
+  name: performance
+spec:
+  cpu:
+    isolated: "0-15"
+    reserved: "0-7"
```

Comment:
isolated and reserved CPUs should not be overlapped. For 16 CPUs it would be good to have, for example:

```
isolated: "2-15"
reserved: "0-1"
```